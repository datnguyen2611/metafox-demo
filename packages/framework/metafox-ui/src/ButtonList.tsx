import { Box, BoxProps } from '@mui/material';
import { styled } from '@mui/material/styles';

type Props = BoxProps & {
  variant?: 'fillFirst' | 'fullWidth' | 'fillWidth';
  spacing?: 'medium' | 'none' | 'small' | 'large';
  fullWidth?: boolean;
};

const ButtonList = styled(
  Box,
  {}
)<Props>(({ theme, spacing, variant, fullWidth }) => ({
  display: 'flex',
  '& button + button': {
    marginLeft: theme.spacing(1)
  },
  ...(variant === 'fillFirst' && {
    flex: 1,
    '& button:nth-of-type(1)': {
      flex: 1
    }
  }),
  ...(variant === 'fullWidth' && {
    flex: 1
  }),
  ...(variant === 'fillWidth' && {
    '& button': {
      flexGrow: 1
    }
  }),
  ...(spacing === 'small' && {
    marginTop: theme.spacing(1)
  }),
  ...(spacing === 'medium' && {
    marginTop: theme.spacing(1.5)
  }),
  ...(spacing === 'large' && {
    margin: theme.spacing(1.5)
  })
}));

export default ButtonList;
