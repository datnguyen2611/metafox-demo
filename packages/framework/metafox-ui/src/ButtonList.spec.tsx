import { render } from '@testing-library/react';
import * as React from 'react';
import ButtonList from './ButtonList';

const MockChildrenComponent = () => (
  <div data-testid="children">content children</div>
);

describe('ButtonList', () => {
  it('render ButtonList', () => {
    const { container } = render(<ButtonList />);
    expect(container).toMatchSnapshot();
  });
  it('render prop variant', () => {
    const props = {
      variant: 'fillFirst'
    };
    const { container, queryByTestId } = render(
      <ButtonList {...props}>
        <MockChildrenComponent />
      </ButtonList>
    );
    expect(queryByTestId('children')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('render prop variant medium', () => {
    const props = {
      variant: 'fillFirst',
      spacing: 'medium'
    };
    const { container, queryByTestId } = render(
      <ButtonList {...props}>
        <MockChildrenComponent />
      </ButtonList>
    );
    expect(queryByTestId('children')).toBeInTheDocument();
    expect(queryByTestId('children')).toHaveTextContent('content children');
    expect(container).toMatchSnapshot();
  });
});
