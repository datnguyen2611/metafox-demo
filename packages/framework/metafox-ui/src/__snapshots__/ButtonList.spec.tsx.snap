// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonList render ButtonList 1`] = `
<div>
  <div
    class="MuiBox-root css-zdhto5"
  />
</div>
`;

exports[`ButtonList render prop variant 1`] = `
<div>
  <div
    class="MuiBox-root css-eeu4j0"
    variant="fillFirst"
  >
    <div
      data-testid="children"
    >
      content children
    </div>
  </div>
</div>
`;

exports[`ButtonList render prop variant medium 1`] = `
<div>
  <div
    class="MuiBox-root css-wa8404"
    spacing="medium"
    variant="fillFirst"
  >
    <div
      data-testid="children"
    >
      content children
    </div>
  </div>
</div>
`;
